### PR TITLE
Little fix in highlight in CGarrisonSlot.

### PR DESCRIPTION
**Pull request by FeniksFire**
When someone turn on split mode and after that turn off it, unit isn't highlight in graphics although it is selected. Well I repaired it.

### CODE REVIEW

File: client/widgets/CGarrisonInt.cpp [hunk 1]
```diff
@@ -514,11 +514,14 @@ void CGarrisonInt::setSplittingMode(bool on)
 
 	if (inSplittingMode || on)
 	{
-		for(CGarrisonSlot * slot : slotsUp)
-			slot->setHighlight( ( on && (slot->our() || slot->ally()) && (slot->creature == nullptr || slot->creature == getSelection()->creature)));
-
-		for(CGarrisonSlot * slot : slotsDown)
-			slot->setHighlight( ( on && (slot->our() || slot->ally()) && (slot->creature == nullptr || slot->creature == getSelection()->creature)));
+		for(CGarrisonSlot * slot : slotsUp){
```
Comment (DjWarmonger) [inline client/widgets/CGarrisonInt.cpp]: No braces at the end of line, please. http://wiki.vcmi.eu/index.php?title=Coding_guidelines#Where_to_put_braces

Comment (FeniksFire) [inline client/widgets/CGarrisonInt.cpp]: Done.

File: client/widgets/CGarrisonInt.cpp [hunk 2]
```diff
@@ -515,10 +515,15 @@ void CGarrisonInt::setSplittingMode(bool on)
...
+			if(slot!=getSelection())
+				slot->setHighlight( ( on && (slot->our() || slot->ally()) && (slot->creature == nullptr || slot->creature == getSelection()->creature)));
+		}
 		for(CGarrisonSlot * slot : slotsDown)
-			slot->setHighlight( ( on && (slot->our() || slot->ally()) && (slot->creature == nullptr || slot->creature == getSelection()->creature)));
+		{
+			if(slot!=getSelection())
+				slot->setHighlight( ( on && (slot->our() || slot->ally()) && (slot->creature == nullptr || slot->creature == getSelection()->creature)));
+		}
 		inSplittingMode = on;
```
Comment (alexvins) [inline client/widgets/CGarrisonInt.cpp]: Looks like 2 cycles are identical inside. Could you simplify this?

Comment (FeniksFire) [inline client/widgets/CGarrisonInt.cpp]: Done.

**Pull request merged**